arquillian-chameleon-runner/runner/src/main/java/org/arquillian/container/chameleon/runner/ArquillianChameleon.java

```diff
@@ -66,9 +65,9 @@ private boolean isInClientSide(ClassLoader parent) {
     }
 
     private void addsArquillianFile(Path file, ClassLoader classLoader) throws Exception {
-        Method method = URLClassLoader.class.getDeclaredMethod("addURL", new Class[] {URL.class});
-        method.setAccessible(true);
-        method.invoke(classLoader, new Object[] {file.toUri().toURL()});
+        final URL url = file.toUri().toURL();
+        final ClassLoader wrappedClassLoader = URLClassLoader.newInstance(new URL[]{url}, classLoader);
+        Thread.currentThread().setContextClassLoader(wrappedClassLoader);
```

Comment:
@lordofthejars this particular change unblocks Chameleon on JDK9+, as starting from this release system classloader is not anymore a subclass of `URLClassLoader`. That was leading to runtime errors while trying to do that reflective call.

Reply:
A bit more of the technical details can be found here https://community.oracle.com/thread/4011800